src/modules/rtp_media_server/rtp_media_server.c

```diff
@@ -0,0 +1,716 @@
+ * PROC_TIMER    -1  Timer attendant process
+ * PROC_FIFO     -2  FIFO attendant process
+ * PROC_TCP_MAIN -4  TCP main process
+ * PROC_UNIXSOCK -5  Unix domain socket server processes
+ *
+ * If this function returns a nonzero value the loading of OpenSER will
+ * stop.
+ */
+static int child_init(int rank)
+{
+	if(rank == PROC_MAIN) {
+		int pid;
+		pid = fork_process(PROC_XWORKER, "RTP_media_server", 1);
+		if(pid < 0)
+			return -1;
+		if(pid == 0) {
+			rms_session_manage_loop();
+			return 0;
+		}
+	}
+	int rtn = 0;
+	return (rtn);
+}
+
+int rms_str_dup(str *dst, str *src, int shared)
+{
+	if(!dst) {
+		LM_ERR("dst null\n");
+		return -1;
+	}
+	dst->len = 0;
+	dst->s = NULL;
+	if(!src) {
+		LM_ERR("src null\n");
+		return 0;
+	}
+	if((!src->s) || (src->len < 1)) {
+		LM_ERR("empty src\n");
+		return 0;
+	}
+	if(shared) {
+		dst->s = shm_malloc(src->len + 1);
+	} else {
+		dst->s = pkg_malloc(src->len + 1);
+	}
+	if(!dst->s) {
+		LM_ERR("%s_malloc: can't allocate memory (%d bytes)\n",
+				shared ? "shm" : "pkg", src->len);
+		return -1;
+	}
+	strncpy(dst->s, src->s, src->len);
+	dst->s[src->len] = '\0';
+	dst->len = src->len;
+	return 1;
+}
+
+static int rms_get_sdp_info(rms_sdp_info_t *sdp_info, struct sip_msg *msg)
+{
+	sdp_session_cell_t *sdp_session;
+	sdp_stream_cell_t *sdp_stream;
+	str media_ip, media_port;
+	int sdp_session_num = 0;
+	int sdp_stream_num = get_sdp_stream_num(msg);
+	if(parse_sdp(msg) < 0) {
+		LM_INFO("can not parse sdp\n");
+		return 0;
+	}
+	sdp_info_t *sdp = (sdp_info_t *)msg->body;
+	if(!sdp) {
+		LM_INFO("sdp null\n");
+		return 0;
+	}
+	rms_str_dup(&sdp_info->recv_body, &sdp->text, 1);
+	if(!sdp_info->recv_body.s)
+		goto error;
+	LM_INFO("sdp body - type[%d]\n", sdp->type);
+	if(sdp_stream_num > 1 || !sdp_stream_num) {
+		LM_INFO("only support one stream[%d]\n", sdp_stream_num);
+	}
+	sdp_stream_num = 0;
+	sdp_session = get_sdp_session(msg, sdp_session_num);
+	if(!sdp_session) {
+		return 0;
+	} else {
+		int sdp_stream_num = 0;
+		sdp_stream = get_sdp_stream(msg, sdp_session_num, sdp_stream_num);
+		if(!sdp_stream) {
+			LM_INFO("can not get the sdp stream\n");
+			return 0;
+		} else {
+			rms_str_dup(&sdp_info->payloads, &sdp_stream->payloads, 1);
+			if(!sdp_info->payloads.s)
+				goto error;
+		}
+	}
+	if(sdp_stream->ip_addr.s && sdp_stream->ip_addr.len > 0) {
+		media_ip = sdp_stream->ip_addr;
+	} else {
+		media_ip = sdp_session->ip_addr;
+	}
+	rms_str_dup(&sdp_info->remote_ip, &media_ip, 1);
+	if(!sdp_info->remote_ip.s)
+		goto error;
+	rms_str_dup(&media_port, &sdp_stream->port, 0);
+	if(!media_port.s)
+		goto error;
+	sdp_info->remote_port = atoi(media_port.s);
+	pkg_free(media_port.s);
+	return 1;
+error:
+	rms_sdp_info_free(sdp_info);
+	return 0;
+}
+
+static int rms_relay_call(struct sip_msg *msg)
+{
+	if(!tmb.t_relay(msg, NULL, NULL)) {
+		LM_INFO("t_ralay error\n");
+		return -1;
+	}
+	return 1;
+}
+
+str reply_headers = {0, 0};
+str headers = str_init("Max-Forwards: 70" CRLF);
```

Comment:
This and the following vars are only used in one functions. Please consider moving them to the respective functions, they don't need to be globally visible.